src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -101,6 +101,25 @@ public ScriptFile GetFile(string filePath)
             return scriptFile;
         }
 
+        /// <summary>
+        /// Tries to get an open file in the workspace. Returns true or false if it succeeds.
+        /// </summary>
+        /// <param name="filePath">The file path at which the script resides.</param>
+        /// <param name="scriptFile">The out parameter that will contain the ScriptFile object.</param>
+        public bool TryGetFile(string filePath, out ScriptFile scriptFile)
+        {
+            try
+            {
+                scriptFile = GetFile(filePath);
+                return true;
+            }
+            catch (FileNotFoundException)
```

Comment:
I'm guessing there are other reasons for failure here, like in:
https://github.com/PowerShell/PowerShellEditorServices/blob/13f21070cdc6075aeb8ab853fafa8f0255bc6aa1/src/PowerShellEditorServices/Language/LanguageService.cs#L345-L355

It looks like we need to consolidate all our file access into a single internal API.

Reply:
I think we should leave `GetFile()`, move this large catch inside `TryGetFile()`, and reuse `TryGetFile()` in the `LanguageService` and `DebugAdapter`. It looks like the opposite consideration is in `GetFile()` itself:
https://github.com/PowerShell/PowerShellEditorServices/blob/834ae3dbe60dcda881d350e003f597f3620ab4a0/src/PowerShellEditorServices/Workspace/Workspace.cs#L81-L98

Reply:
I want to call this out. 

We are logging the exception as well in the current implementation. Typically I've not seen `TryX` functions give access to the underlying exception.

That said, I guess we could add an overload that will let you out the exception as well:

```
public bool TryGetFile(string filePath, out ScriptFile scriptFile, out Exception exception)
```

Reply:
I assume we don't want to log the exception within the `Try_` method? It would contain a full stack trace, but we could log it as a warning, since we used a `Try_`.